tools/tsTreeshaker.ts

```diff
@@ -0,0 +1,419 @@
+// code based on ts-transform-paths
+import fs from 'fs';
+import path from 'path';
+import ts, { factory } from 'typescript';
+
+class DependencyGraph {
+  private readonly graph: Map<string, Set<string>> = new Map();
+  private readonly includedDependencies = new Set<string>();
+  private graphNeedCompute = true;
+  private readonly sourceRoots: readonly string[];
+  constructor(sourceRoots: readonly string[]) {
+    this.sourceRoots = sourceRoots;
+  }
+  private computeDependencies() {
```

Comment:
It seems like you forget to set `graphNeedCompute` to `false`?

Reply:
I'm doing it in line 16